jazkarta/shop/browser/checkout.py

```diff
@@ -24,13 +32,318 @@
 from ..vocabs import country_names
 
 
-@implementer(IStripeEnabledView)
 class CheckoutForm(BrowserView):
+
+    def __call__(self):
+        payment_processor = get_setting('payment_processors')
+        if payment_processor != []:
+            p_p = payment_processor[0]
+            if p_p == 'Authorize.Net SIM':
+                return CheckoutFormAuthorizeNetSIM(self.context, self.request)()
+            elif p_p == 'Stripe':
+                return CheckoutFormStripe(self.context, self.request)()
+        else:
+            raise Exception(
+                    'No payment processor has been specified.')
+
+
+class CheckoutFormAuthorizeNetSIM(BrowserView):
+    """ Renders a checkout form with button to submit to Authorize.Net SIM """
+
+    cart_template = ViewPageTemplateFile('templates/checkout_cart.pt')
+    thankyou_template = ViewPageTemplateFile('templates/checkout_thankyou.pt')
+    index = ViewPageTemplateFile('templates/checkout_form_authorize_net_sim.pt')
+    receipt_email = ViewPageTemplateFile('templates/receipt_email.pt')
+    post_url_test = 'https://test.authorize.net/gateway/transact.dll'    
+    post_url_production = 'https://secure.authorize.net/gateway/transact.dll'
+
+    # Note: Customer only has 15min to complete the checkout process as per:
+    # https://support.authorize.net/authkb/index?page=content&id=A592&actp=LIST
+
+    def __call__(self):
+        if 'x_response_code' in self.request.form:
+            # recreate the cart from session or from storage (by user_id)
+            user_id = self.request.form.get('user_id', None)
+            session_id = self.request.form.get('session_id', None)
+            self.cart = Cart.from_session_id(self.request, user_id, session_id)
+        self.update()
+        if 'x_response_code' in self.request.form:
+            self.handle_submit()
+        return self.render()
+
+    def update(self):
+        self.error = None
+        try:
+            self.cart.calculate_taxes()
+        except TaxRateException:
+            # The sales tax could not be calculated as some of the
+            # required information was missing. (can be triggered by anon users,
+            # hitting the checkout url directly)
+            return 
```

Comment:
I think doing no exception handling is the best thing here. We anticipate that there will be conditions where we can't calculate the tax but we don't yet understand what they are. With no exception handling, it'll raise an exception, and we should get an email (at least, I would on sites that I run, because I set up mailinglogger). If we catch the exception and tell the user that there was a problem calculating tax, that's a bit nicer for the user, but then we lose the traceback so it'll be harder to figure out what happened.